compiler/noirc_frontend/src/parser/parser/expression.rs

```diff
@@ -855,8 +855,11 @@ impl Parser<'_> {
     }
 
     fn parse_statement_in_block(&mut self) -> Option<(Statement, (Option<Token>, Location))> {
-        if let Some(statement) = self.parse_statement() {
-            Some(statement)
+        if let Some((statement, semicolon, opt_cfg_attribute)) = self.parse_statement() {
+            if !self.is_enabled_cfg(opt_cfg_attribute) {
+                return None;
+            }
+            Some((statement, semicolon))
```

Comment:
I'm curious why the parser has knowledge of cfg attributes - why not handle this logic in the elaborator for example?